unblob/parser.py

```diff
@@ -0,0 +1,87 @@
+import itertools
+
+from lark import Lark, Transformer, UnexpectedCharacters
+
+_hex_string_parser = Lark(
+    """
+    %import common.HEXDIGIT
+    %import common.NUMBER
+    %import common.NEWLINE
+    %import common.WS
+    %ignore WS
+
+    COMMENT:  "//" /(.)+/ NEWLINE?
+
+    %ignore COMMENT
+
+    start: item+
+
+    item: LITERAL           -> literal
+        | WILDCARD          -> wildcard
+        | FIRSTNIBLE        -> first_nible
+        | SECONDNIBLE       -> second_nible
+        | JUMP              -> jump
+        | RANGE_JUMP        -> range_jump
+        | alternative
+
+    alternative: "(" item (ALTERNATIVE_SEPARATOR item+)+ ")"
```

Comment:
The first alternative can not be a sequence of e.g. `LITERAL`s - why do we have this limitation?

For consistency, I would also prefer `"|"` here instead of `ALTERNATIVE_SEPARATOR`, as we already have `"("` and `")"`.

Reply:
You are right about the first, that was a typo, added a testcase and fixed it.

`"|"` would be swallowed by the parser, while `ALTERNATIVE_SEPARATOR` is a terminal and hence part of the parser result, which can be used by the Transformer visitor to group items together.